### PR TITLE
Add assertion type to the results returned by each assert

### PR DESCRIPTION
**Pull request by ArianJM**
Adds a property `assertionType` to the values returned by any assertion.

### CODE REVIEW

File: src/assert.js [hunk 1]
```diff
@@ -414,7 +430,8 @@ class Assert {
 					result,
 					actual,
 					expected,
-					message
+					message,
+					assertionType: assertionType
```
Comment (adrianlizarraga) [inline src/assert.js]: Nitpick: just `assertionType` instead of `assertionType: assertionType` to be consistent

**Review by adrianlizarraga — APPROVED** [review]
Nice! Just one really minor thing

**Comment by AlexEdgcomb** [comment]
@ArianJM , this covers all but 5 asserts (http://api.qunitjs.com/assert/).

3 of which are configurations, so not as important to get the assertType (perhaps):
* async
* expect
* timeout

However, 2 of which seem to be worth collecting the assertType:
* pushResult
* verifySteps

Should we be collecting `pushResult` or `verifySteps`? Note: We can do that in a later PR, but might be nice to get all that done now, if not too much trouble.

**Comment by ArianJM** [comment]
@adrianlizarraga comment addressed.
Also fixed some tests I didn't see I was breaking.

@AlexEdgcomb as I understood it, those do not qualify for adding `assertType`:
* As you say: `async`, `expect` and `timeout` are not really assertions.
  * [`async`](https://api.qunitjs.com/assert/async) works with other assertions, just `async` doesn't assert anything.
  * [`expect`](https://api.qunitjs.com/assert/expect) is used to indicate how many assertions to expect, just `expect` doesn't assert anything. 
  * [`timeout`](https://api.qunitjs.com/assert/timeout) is a timeout for `async` operations. Not really asserting.
* [`pushResult`](https://api.qunitjs.com/assert/pushResult) just reports the result of an assertion. This is actually the method that's being called all the time to report the result, expected, actual, message, etc. for each assertion, in each `assert` method I've added `assertType` to its call to `pushResult` this is not an actual assertion.
* [`verifySteps`](https://api.qunitjs.com/assert/verifySteps) I'm not sure what does this do, I think this ensures that a set of assertions are executed in certain order. It's not an actual assertion either.

Let me know if something doesn't add up, or I'm wrong.

**Comment by AlexEdgcomb** [comment]
@ArianJM , thanks!

I think `async` and `expect` may only be `log`ged when something goes wrong, so perhaps our `log` function can figure out the particular assertion, then report important info:
* `async`: How long was the timeout in ms? 5000? (if possible)
* `expect`: What was the expected number of asserts and what was the actual?

However, `pushResult` and `verifySteps` always get `log`ged, so I think we'll want to capture those like the ones you've already captured (if possible):
* `pushResult`: Maybe adjust `pushResult` function to set an `assertType` to `pushResult` if no other `assertType` is set?
* `verifySteps`: Actual/expected are an array of strings. Ex:
```
assert.step('test') // logged
assert.step('test2') // logged
assert.verifySteps([ 'test', 'test2' ]) // logged
```

**Comment by ArianJM** [comment]
@AlexEdgcomb added `pushResult` and `verifySteps`.

> I think `async` and `expect` may only be `log`ged when something goes wrong, so perhaps our `log` function can figure out the particular assertion, then report important info:
> * `async`: How long was the timeout in ms? 5000? (if possible)
> * `expect`: What was the expected number of asserts and what was the actual?

Done here https://github.com/zyBooks/zydeSDK/pull/91/commits/f9d680cd4d62c46dc273c9e2bfc41e6168fb42ac#diff-e365e7f9f2715cc7db410cc7039f0252

**Review by AlexEdgcomb — APPROVED** [review]
@ArianJM , outstanding :D Approved.

**Comment by ArianJM** [comment]
@AlexEdgcomb probably ready to merge.

**Pull request merged**